tool/tsh/common/tsh.go

```diff
@@ -2555,49 +2555,72 @@ func onLogout(cf *CLIConf) error {
 
 	switch {
 	// Proxy and username for key to remove.
-	case proxyHost != "" && cf.Username != "":
-		tc, err := makeClient(cf)
-		if err != nil {
-			return trace.Wrap(err)
-		}
-		// Load profile for the requested proxy/user.
-		profile, err := tc.ProfileStatus()
-		if err != nil && !trace.IsNotFound(err) && !trace.IsCompareFailed(err) {
-			return trace.Wrap(err)
+	case proxyHost != "":
+		// In the event --user flag is not supplied, and there is only one identity,
+		// we can simply log out the single identity.
+		if cf.Username == "" {
+			clientStore := cf.getClientStore()
+			usernames, err := clientStore.GetIdentities(proxyHost)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+
+			logger.DebugContext(cf.Context, "No --user flag provided, but identities found for proxy",
+				"proxy_host", proxyHost,
+				"users", usernames)
+
+			if len(usernames) == 1 {
+				cf.Username = usernames[0]
+			}
```

Comment:
This whole onLogout function is pretty messy with long "case" branches and now several levels of indentation which IMO hurts readability a bit.

What do you think about the following: here, if we found more than 1 identity, print the message saying you need to provide --user (which you have way down below) and return. Then the switch below that checks on cf.Username != "" won't be needed, the whole thing will unintend back to the left and the diff will be smaller/cleaner too.

WDYT?